ts/features/wallet/onboarding/paypal/saga/index.ts

```diff
@@ -0,0 +1,76 @@
+import { call, put, takeLatest } from "redux-saga/effects";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import { NonNegativeNumber } from "@pagopa/ts-commons/lib/numbers";
+import { PaymentManagerClient } from "../../../../../api/pagopa";
+import { SessionManager } from "../../../../../utils/SessionManager";
+import { PaymentManagerToken } from "../../../../../types/pagopa";
+import { searchPaypalPsp } from "../store/actions";
+import { getGenericError, getNetworkError } from "../../../../../utils/errors";
+import { SagaCallReturnType } from "../../../../../types/utils";
+import { PayPalPsp as NetworkPsp } from "../../../../../../definitions/pagopa/PayPalPsp";
+import { IOPayPalPsp } from "../types";
+import { getPayPalPspIconUrl } from "../../../../../utils/paymentMethod";
+
+// convert a paypal psp returned by the API into the app domain model
+const convertNetworkPsp = (psp: NetworkPsp): IOPayPalPsp => ({
+  id: psp.idPsp,
+  logoUrl: getPayPalPspIconUrl(psp.codiceAbi),
+  name: psp.ragioneSociale,
+  fee: psp.maxFee as NonNegativeNumber,
+  privacyUrl: psp.privacyUrl,
+  // TODO see https://www.figma.com/file/YokChZzMKIaQ4ouWaS22En?node-id=333:7168#122840225
+  tosUrl: ""
+});
+
+// handle the request of search for PayPal psp
+function* handlePaypalSearchPsp(
+  sarchPsp: PaymentManagerClient["searchPayPalPsp"],
+  sessionManager: SessionManager<PaymentManagerToken>
+) {
+  try {
+    const searchPayPalPspRequest: SagaCallReturnType<typeof sarchPsp> =
+      yield call(sessionManager.withRefresh(sarchPsp));
```

Comment:
Since if the request fails the user can retry pressing the retry button, what do you think about protect the API call with exponential backoff? 

https://user-images.githubusercontent.com/11773070/141455757-acde6a37-310b-48a1-b0e9-2200efbec276.mov

Reply:
Yes, I believe it is a good idea!
Since it is not in the scope of this task, I created a dedicated one https://pagopa.atlassian.net/browse/IA-449